src/cursor/abstract_cursor.ts

```diff
@@ -457,9 +458,24 @@ export abstract class AbstractCursor<
    * cursor.rewind() can be used to reset the cursor.
    */
   async toArray(): Promise<TSchema[]> {
-    const array = [];
+    const array: TSchema[] = [];
+
+    // when each loop iteration ends,documents will be empty and a 'await (const document of this)' will run a getMore operation
     for await (const document of this) {
       array.push(document);
+      let docs = this.readBufferedDocuments();
+      if (this.transform != null) {
+        docs = await Promise.all(
+          docs.map(async doc => {
```

Comment:
After looking into Bailey's comment on removing the need to await a promise, if we still have `await` I'm not sure if it is desirable for us to create all these promises without awaiting them, and then in parallel find out if one of them threw. We should probably change this into a loop proper, rather than an `.all(.map)` style call.

Reply:
Done!